src/status.rs

```diff
@@ -0,0 +1,738 @@
+            heartbeats: None,
+        };
+
+        server.expect(Expectation::matching(any()).respond_with(json_encoded(response)));
+
+        let sentry = SentryApi::new().expect("Should work");
+
+        let actual = is_finalized(&sentry, &channel)
+            .await
+            .expect("Should query dummy server");
+        let expected = IsFinalized::Yes {
+            reason: Finalized::Withdraw,
+            balances: Default::default(),
+        };
+
+        assert_eq!(expected, actual);
+    }
+
+    #[test]
+    fn is_offline_no_heartbeats() {
+        let messages = Messages {
+            leader: LastApprovedResponse {
+                last_approved: None,
+                heartbeats: Some(vec![]),
+            },
+            follower: LastApprovedResponse {
+                last_approved: None,
+                heartbeats: Some(vec![]),
+            },
+            recency: Duration::minutes(4),
+        };
+
+        assert!(
+            is_offline(&messages),
+            "On empty heartbeat it should be offline!"
+        );
+
+        let messages = Messages {
+            leader: LastApprovedResponse {
+                last_approved: None,
+                heartbeats: None,
+            },
+            follower: LastApprovedResponse {
+                last_approved: None,
+                heartbeats: Some(vec![]),
+            },
+            recency: Duration::minutes(4),
+        };
+
+        assert!(
+            is_offline(&messages),
+            "On empty heartbeat it should be offline!"
+        );
+    }
+
+    // is_date_recent()
+    #[test]
+    fn now_date_is_recent() {
+        let now = Utc::now();
+        let recency = Duration::minutes(4);
+        assert!(
+            is_date_recent(recency, &now),
+            "The present moment is a recent date!"
+        )
+    }
+
+    #[test]
+    fn slightly_past_is_recent() {
+        let recency = Duration::minutes(4);
+        let on_the_edge = Utc::now().checked_sub_signed(Duration::minutes(3)).unwrap();
+        assert!(
+            is_date_recent(recency, &on_the_edge),
+            "When date is just as old as the recency limit, it still counts as recent"
+        )
+    }
+
+    #[test]
+    fn old_date_is_not_recent() {
+        let recency = Duration::minutes(4);
+        let past = Utc::now() - Duration::minutes(10);
+        assert_eq!(
+            is_date_recent(recency, &past),
+            false,
+            "Date older than the recency limit is not recent"
+        )
+    }
+
+    // is_initializing()
+    #[test]
+    fn two_empty_message_arrays() {
+        let messages = Messages {
+            leader: LastApprovedResponse {
+                last_approved: None,
+                heartbeats: Some(vec![]),
+            },
+            follower: LastApprovedResponse {
+                last_approved: None,
+                heartbeats: Some(vec![]),
+            },
+            recency: Duration::minutes(4),
+        };
+
+        assert_eq!(
```

Comment:
Note: You can use `assert!` (it's checking for True values), but no need to change it.